weave/index/index_sql.py

```diff
@@ -59,22 +51,29 @@ def __init__(self, file_system, pantry_path, **kwargs):
             # even though they are not used until we connect.
             # Pylint thinks this is pointless, so pylint is getting ignored.
             # pylint: disable=pointless-statement
-            os.environ["MSSQL_HOST"]
-            os.environ["MSSQL_USERNAME"]
-            os.environ["MSSQL_PASSWORD"]
+            self._sql_connection = {
+                'host': os.environ["WEAVE_SQL_HOST"],
+                'username': os.environ["WEAVE_SQL_USERNAME"],
+                'password': os.environ["WEAVE_SQL_PASSWORD"],
+                'port': os.environ.get("WEAVE_SQL_PORT", 5432),
+            }
             # pylint: enable=pointless-statement
         except KeyError as key_error:
             raise KeyError("The following environment variables must be set "
-                           "to use this class: MSSQL_HOST, MSSQL_USERNAME, "
-                           "MSSQL_PASSWORD.") from key_error
+                           "to use this class: WEAVE_SQL_HOST, "
+                           "WEAVE_SQL_USERNAME, WEAVE_SQL_PASSWORD."
+                          ) from key_error
 
         self._file_system = file_system
         self._pantry_path = pantry_path
 
         # Set the database name (defaults to weave_db). DATABASE MUST ALREADY
         # EXIST. If it does not, the user must create it manually.
-        self._database_name = kwargs.get("database_name", "weave_db")\
-            .replace("-", "_")
+        self._database_name = kwargs.get(
```

Comment:
I know you didn't write this comment, but what happens if the database doesn't already exist?

Reply:
If it doesn't exist, we get a hard failure from the sqla.create_engine call on line 86. This is not something that I think we can (or should) create in the python, we just want to connect to something that exists. Typically, you need to create the database outside of Python as an admin.